cpp/src/reshape/interleave_columns.cu

```diff
@@ -140,85 +141,53 @@ struct interleave_columns_impl<T, std::enable_if_t<std::is_same_v<T, cudf::struc
   }
 };
 
+struct interleave_strings_fn {
+  using string_index_pair = cudf::strings::detail::string_index_pair;
+  table_device_view d_table;
+
+  __device__ string_index_pair operator()(size_type idx)
+  {
+    auto const num_columns    = d_table.num_columns();
+    auto const source_col_idx = idx % num_columns;
+    auto const source_row_idx = idx / num_columns;
```

Comment:
This access pattern makes me wonder if a kernel would be significantlty faster.
But I assume this is light-weigth either way.

Reply:
Do you mean faster than `thrust::transform`? 
The lambda here should be very fast since it only operates on the bitmask and the offsets in a very coalesced access pattern.

Reply:
the reason I thought about this is because threads with adjacent indices access different columns.

Reply:
Ah yes. That is a good point.

Reply:
I almost made the same comment as @vuule but then I wondered if the point was that interleaving would have coalesced writes (not reads)? I didn’t look too closely but my intuition was that swapping these might be worthwhile. At least worth benchmarking.

Reply:
I switched the order from coalesced write to coalesced read and wrote a benchmark with different number of columns. The performance did suffer 10% (for 2 columns) to 35% (for 100 columns).
This probably could be mitigated with some extra work to use shared-memory to minimize the coalesced writes.
But I think this kind of effort should also encompass the non-strings code paths as well (which also do coalesced writes). So I feel this may be a bit out of scope for this PR.
I will include the benchmark code in this PR since it has already been created.

Reply:
Amazing. Thanks for measuring.